File: client/src/schema/taskControlState.ts
import type { MergeCoreCollection } from '../index.js';

/**
 * vytex_task_control_status type
 */
export type VytexTaskControlState<Schema = any> = MergeCoreCollection<
	Schema,
	'vytex_orders-status',
	{
		id: number;
		name: string;
		value: TaskControlStateValue;
	}
>;

export type TaskControlStateValue = 'created' | 'started' | 'finished' | 'rejected';

File: api/app/v1/models/task_control_state.go
package models

type TaskControlState struct {
	ID    uint                  `json:"id,omitempty" gorm:"primary_key"`
	Name  string                `json:"name"`
	Value TaskControlStateValue `json:"value" gorm:"type:text"`
}

type TaskControlStateValue string

const (
	CreatedTaskControlStateValue  = "created"
	StartedTaskControlStateValue  = "started"
	RejectedTaskControlStateValue = "rejected"
	FinishedTaskControlStateValue = "finished"
)

func DefaultTaskControlStatus() []*TaskControlState {
	return []*TaskControlState{
		{Name: "Creada", Value: CreatedTaskControlStateValue},
		{Name: "Iniciada", Value: StartedTaskControlStateValue},
		{Name: "Rechazada", Value: RejectedTaskControlStateValue},
		{Name: "Finalizada", Value: FinishedTaskControlStateValue},
	}
}
